Codes/coordinate_initialization.py

```diff
@@ -0,0 +1,228 @@
+global arduino_confirm
+global comm_string
+global ser
+global line_in
+global phase  # phase 1 : for red border and phase 2 : for black border
+global distance
+global found_red
+global found_black
+global mm_per_pix
+global coord_ini_status
+
+#************Initialization*************
+
+#ser = serial.Serial('/dev/ttyACM0',9600)
+phase=1
+found_red=False
+found_black=False
+dist_y_bord=20  #mm
+dist_x_bord=20  #mm
+mm_per_pix=0.364996557
+cap = cv2.VideoCapture(0)
+cap.set(3,1280)
+cap.set(4,1024)
+time.sleep(2)
+coord_ini_status=False
+
+#**************************************
+#************Functions******************
+
+#****************************************
+#Function Name:get_ratio_points
+#  * Input: coordinates of two points and ratios
+#  * Output: required coordinate
+#			dividing line into req. ratio	
+#  * Logic: line ratio formula
+#  * Example Call: get_ratio_points(1,1,6,6,1,2)
+#****************************************
+def get_ratio_points(x1,x2,y1,y2,k1,k2):
+  x0 = ((x1*k2)+(x2*k1))/(k1+k2)
+  y0 = ((y1*k2)+(y2*k1))/(k1+k2)
+  return int(x0),int(y0)
+
+#****************************************
+#Function Name:cal_distance_for_red
+#  * Input: 4 coord. of border red 
+#  * Output: none
+#  * Logic: get center line of red border
+#			cal. the distance from midpoint of photo 
+#  * Example Call: cal_distance_for_red(1,2,3,4,5,6,7,8)
+#****************************************
+def cal_distance_for_red(x1,x2,x3,x4,y1,y2,y3,y4):
+	xc,yc=(640,512)
+	yr=512
+	xt,yt=get_ratio_points(x1,x2,y1,y2,1,1)
+	xd,yd=get_ratio_points(x2,x3,y2,y3,1,1)
+	m=(yd-yt)/(xd-xt)
+	c=yd-(m*xd)
+	xr=(yr-c)/m;
+	#distance in pixels
+	distance = math.sqrt(math.pow(xc-xr,2)+math.pow(yc-yr,2))
+	distance=distance*mm_per_pix
+
+#****************************************
+#Function Name:cal_distance_for_black
+#  * Input: 4 coord. of border black 
+#  * Output: none
+#  * Logic: get center line of black border
+#			cal. the distance from midpoint of photo 
+#  * Example Call: cal_distance_for_black(1,2,3,4,5,6,7,8)
+#****************************************
+def cal_distance_for_black(x1,x2,x3,x4,y1,y2,y3,y4):
+	xc,yc=(640,512)
+	xr=640
+	xt,yt=get_ratio_points(x1,x3,y1,y3,1,1)
+	xd,yd=get_ratio_points(x2,x4,y2,y4,1,1)
+	m=(yd-yt)/(xd-xt)
+	c=yd-(m*xd)
+	yr=m*xr+c
+	#distance in pixels
+	distance = math.sqrt(math.pow(xc-xr,2)+math.pow(yc-yr,2))
+	distance=distance*mm_per_pix
+
+#****************************************
+#Function Name:go_for_y
+#  * Input: int final-whether to go for final run
+#  * Output: none
+#  * Logic: will generate the serial string
+# 			and communicate to arduino, followed
+#			by confirmation  (for y direction)
+#  * Example Call:go_for_y()
+#****************************************
+def go_for_y(final):
+	# communication string : @ + x_coord + y_coord + z_coord + seed_bit + water_bit + initialization_bit + !
+	arduino_confirm=False
+	mystring=""
+	if final == 0 :
+		comm_string="@"+"0000"+str(dist_y_bord).zfill(3)+"000"+"002"+"!"
```

Comment:
this code seems to be reused in go_for_x you can make seperate function out of it